### ISSUE TITLE
[Bug] CCS routing requires partners to know the spec, which is an unfair requirement to push on them. We need a helper: WithCcsRoutingHint

### ISSUE THREAD

**Issue by jmprieur** [AzureAD/microsoft-authentication-library-for-dotnet#2725]
## Why?
Today, customers writing ASP.NET MVC web apps, need to understand the CCS routing spec, which is not something we want.

## What?
Have a dedicated API `CcsHintExtensions` on the AcquireTokenXXBuilders to pass-in the necessary information (oid, tid, or even better claims)

Example (non intrusive sample using an extension method, to get the idea) 

```CSharp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.

using System.Collections.Generic;
using System.Security.Claims;
using Microsoft.Identity.Client;
using Microsoft.Identity.Client.Advanced;

namespace Microsoft.Identity.Web
{
    /// <summary>
    /// Extension methods to send CCS headers.
    /// </summary>
    public static class CcsHintExtensions
    {
        /// <summary>
        /// Sets the CCS Routing hint.
        /// </summary>
        /// <typeparam name="T">Builder type.</typeparam>
        /// <param name="builder">Builder.</param>
        /// <param name="user">Claims principal for the user.</param>
        /// <returns>The builder to chain.</returns>
        public static AbstractAcquireTokenParameterBuilder<T> WithCcsRoutingHint<T>(
            this AbstractAcquireTokenParameterBuilder<T> builder,
            ClaimsPrincipal? user)
            where T : AbstractAcquireTokenParameterBuilder<T>
        {
            string ccsRoutingHint = CreateCcsRoutingHintFromHttpContext(user);
            builder.WithExtraHttpHeaders(new Dictionary<string, string>
            {
                { Constants.XAnchorMailbox, ccsRoutingHint },
            });

            return builder;
        }

        private static string CreateCcsRoutingHintFromHttpContext(ClaimsPrincipal? user)
        {
            if (user != null)
            {
                string? oid = user.GetObjectId();
                string? tid = user.GetTenantId();
                if (!string.IsNullOrEmpty(oid) && !string.IsNullOrEmpty(tid))
                {
                    return $"oid:{oid}@{tid}";
                }
            }

            return string.Empty;
        }
    }
}
```

**Comment by bgavrilMS** [comment]
Seems more like a feature request @jmprieur, as you can inject the same information via `WithExtraHttpHeaders`. I agree that a convenience mechanism would be good though, we should take this.

**Comment by jmprieur** [comment]
@bgavrilMS that's not the point. 
I'm fine with having a generic mechanism to inject extra http headers
but we **cannot** ask partners who don't use Ms.Id.Web to do that to optimize the CCS routing. This requires too much knowledge and goes in the way of getting **out of the box** resilience.

**Comment by trwalke** [comment]
released in [4.34.0](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/releases/tag/4.34.0)
